--- a/cctools/libstuff/macosx_deployment_target.c
+++ b/cctools/libstuff/macosx_deployment_target.c
@@ -26,7 +26,9 @@
 #include <string.h>
 #include <mach/mach.h>
 #include <sys/types.h>
+#ifdef __APPLE__
 #include <sys/sysctl.h>
+#endif
 #include "stuff/errors.h"
 #include "stuff/allocate.h"
 #include "stuff/macosx_deployment_target.h"
--- a/cctools/include/foreign/sys/sysctl.h
+++ b/cctools/include/foreign/sys/sysctl.h
@@ -1,4 +1,4 @@
-#ifndef __CYGWIN__
+#if !defined(__CYGWIN__) && !defined(__linux__)
 #include_next <sys/sysctl.h>
 #else
 #ifndef __SYSCTL_H__
